blue_modules/notifications.js

```
@@ -265,89 +245,79 @@ const _setPushToken = async token => {
+            onNotification: async notification => {
+              // Deep clone to avoid modifying the original notification
+              const payload = JSON.parse(JSON.stringify({
+                ...notification,
+                ...notification.data
+              }));
+              
+              if (notification.data?.data) {
+                // Validate data before merging
+                const validData = Object.entries(notification.data.data)
+                  .filter(([_, value]) => value !== null && value !== undefined)
+                  .reduce((acc, [key, value]) => ({ ...acc, [key]: value }), {});
```

Biome

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)